### PR TITLE
Implement calculate merkle root from leaves sync - Closes #6885 

### LINKED ISSUE #6885: Add static version of regular merkle root calculation
### Description
Add below function to the regular merkle tree and expose from index.ts
```
calculateMerkleRootWithLeafs(leafData: Buffer[]): Buffer
```

### Motivation

Currently regular merkle tree class is used for both one time merkle root calculation and consecutive merkle root calculation. However, this function must be `async` to support  saving the merkle tree data  into database.
We want to calculate the merkle root in synchronous function in case of creating transaction root and  asset root.

### Acceptance criteria
- Function is unit tested against the fixture

### PR DESCRIPTION
**Pull request by mitsuaki-u**
### What was the problem?

This PR resolves #6885 

### How was it solved?

- 🌱 Added function 427bdd2808f4c7ab2b56976fe3faf43139a56a8d
- 🌱 Added unit tests 0e7311b2793c1d0bf9fc5484a8f1c82882bd9242

### How was it tested?

Unit tests

### CODE REVIEW

File: elements/lisk-tree/test/merkle_tree/utils.spec.ts [hunk 1]
```diff
@@ -54,4 +54,25 @@ describe('utils', () => {
...
 	});
+
+	describe('calculateMerkleRootWithLeaves', () => {
+		describe('should generate correct merkle root info', () => {
+			for (const test of fixture.testCases) {
+				describe(test.description, () => {
+					const transactionIds = test.input.transactionIds.map(hexString =>
+						Buffer.from(hexString, 'hex'),
+					);
+					const tree = new MerkleTree();
```
Comment (shuse2) [inline elements/lisk-tree/test/merkle_tree/utils.spec.ts:65]: Instead of creating merkle tree, fixture has `transactionMerkleRoot` to compare with

**Review by shuse2 — APPROVED** [review]
Other than usage of fixtures, LGTM

**Review by shuse2 — APPROVED** [review]

File: elements/lisk-tree/test/merkle_tree/utils.spec.ts [hunk 2]
```diff
@@ -54,4 +54,24 @@ describe('utils', () => {
...
+
+	describe('calculateMerkleRootWithLeaves', () => {
+		describe('should generate correct merkle root info', () => {
+			for (const test of fixture.testCases) {
+				describe(test.description, () => {
+					const transactionIds = test.input.transactionIds.map(hexString =>
+						Buffer.from(hexString, 'hex'),
+					);
+
+					beforeAll(async () => {});
```
Comment (shuse2) [inline elements/lisk-tree/test/merkle_tree/utils.spec.ts:66]: beforeAll is no longer used

**Review by mehmetegemen — COMMENTED** [review]
I think we wouldn't mind few instructions of 2 conditionals in the recursive function, LGTM, good job!

**Review by mehmetegemen — APPROVED** [review]

**Pull request merged**